File: frontend(BAD)/code/models/lobbyState.py
from dataclasses import dataclass

from models.battle import Battle
from models.lobbySettings import LobbySettings
from models.player import Player


@dataclass
class LobbyState:
    name: str
    id: str
    players: list
    battles: list
    stage: str
    stageTimer: float
    lobbySettings: LobbySettings

    @classmethod
    def from_dict(cls, d):
        return cls(
            name=d['name'],
            id=d['id'],
            players=[Player.from_dict(p) for p in (d.get('players') or [])],
            battles=[Battle.from_dict(b) for b in (d.get('battles') or [])],
            stage=d['stage'],
            stageTimer=d['stageTimer'],
            lobbySettings=LobbySettings.from_dict(d['lobbySettings']),
        )

File: front-end/src/models/LobbyState.js
import { Player } from './Player.js';
import { LobbySettings } from './LobbySettings.js';

import { ALLSTAGES } from '../helpers/constants.jsx';

export class LobbyState {
    constructor(name, id, players, stage, stageTimer, lobbySettings) {
        this.name = name;
        this.id = id;
        this.players = players;
        this.stage = stage;
        this.stageTimer = stageTimer;
        this.lobbySettings = lobbySettings;
    }

    static fromDict(data) {
        return new LobbyState(
            data.name,
            data.id,
            data.players ? data.players.map(p => Player.fromDict(p)) : [],
            data.stage,
            data.stageTimer,
            data.lobbySettings ? LobbySettings.fromDict(data.lobbySettings) : null
        );
    }
}
